prem_utils/connectors/base.py

```diff
@@ -78,3 +115,125 @@ def finetuning(
 
     def get_finetuning_job(self, job_id) -> dict[str, any]:
         raise NotImplementedError
+
+
+# The reason we have a base class for OpenAI is because this can easily integrate
+# providers which provides services through Open AI's SDK
+class OpenAIBaseConnector(BaseConnector):
```

Comment:
The idea is to extend directly the OpenAIConnector. No need to create an OpenAIBaseConnector. Hierarchy can be as deep as we want.

Reply:
Yes, I was initially confused on this, but it makes sense now.